Feature accepted requests

dtuckersuh opened the pull request:
Addresses Task #49 of Agile Board

- Add 'show accepted only' functionality to Request page that will filter all requests (sent and received) based on whether they were accepted or not

- Port in Rating Modal popup created by @CalebNale to add rating functionality after request is completed

- Add functionality to change 'completed' status of requests and update page to filter out completed requests

dtuckersuh committed:
Add Tutor Requests, Sent Requests components to dashboard

dtuckersuh committed:
Add Tutor Requests, Sent Requests components to dashboard

dtuckersuh committed:
Merge branch 'feature-dashboard-requests' of github.com:dtuckersuh/2021-fall-cs160-team-wanda into feature-dashboard-requests

dtuckersuh committed:
Add request functionality from homepage

dtuckersuh committed:
Add status of requests in Sent Requests component

dtuckersuh committed:
Add hyperlink functionality to dashboard components

- Each Tutor Request component now leads to requests page

- Each Suggested Session component now leads to tutors page

- Each Sent Request component now leads to a dummy link representing the
sent requests page

dtuckersuh committed:
Remove href from Suggested Session component

dtuckersuh committed:
Add Received Requests modal and buttons for tutor requests

dtuckersuh committed:
Fix merge with main branch

dtuckersuh committed:
Fix request link for Sent Requests

dtuckersuh committed:
Add 'Accepted' checkbox to requests page

dtuckersuh committed:
Merge branch 'main' into feature-accepted-requests

dtuckersuh committed:
Add checkbox to Requests page that filters requests based on whether they are accepted or not

dtuckersuh committed:
Merge branch 'main' into feature-accepted-requests

dtuckersuh committed:
Style checkbox

dtuckersuh committed:
Merge branch 'main' of github.com:dtuckersuh/2021-fall-cs160-team-wanda into feature-accepted-requests

dtuckersuh committed:
Add Session Rating modal to Request Page

dtuckersuh committed:
Merge branch 'main' of github.com:dtuckersuh/2021-fall-cs160-team-wanda into feature-accepted-requests

dtuckersuh committed:
Add rating popup to complete session

dtuckersuh committed:
Merge branch 'main' of github.com:dtuckersuh/2021-fall-cs160-team-wanda into feature-accepted-requests

dtuckersuh committed:
Fix styling errors and add 'completed' field to received/sent requests state in /requests page

dtuckersuh committed:
Fix status of request

dtuckersuh committed:
Add functionality to change request 'completed' status and remove completed requests from page

thivnguyen commented on tutor4points/core/templates/home.html:
```diff
@@ -44,14 +44,21 @@ <h2>Tutor Requests</h2>
+                                        {% if tutor_request.accepted %}
+                                            <button type="submit" name="btn-accept-request" cless="btn-secondary"
+                                                data-toggle="modal" data-target="#modal-complete-request">
+                                                Complete 
```
I think a better button name would be "Completed and Paid" or something along that line, so the tutor knows that they should only click the button after they have fully complete and paid for their session. Another option is to just add a checkbox to the form as "Paid" and "Completed" and make sure the user check marks those 2 fields before submitting the form.

thivnguyen commented on tutor4points/core/templates/home.html:
```diff
@@ -44,14 +44,21 @@ <h2>Tutor Requests</h2>
-                                        </button>
+                                        {% if tutor_request.accepted %}
+                                            <button type="submit" name="btn-accept-request" cless="btn-secondary"
+                                                data-toggle="modal" data-target="#modal-complete-request">
```
It'll be helpful to still have the status next to the button or above it, so the user knows the request has been accepted.

thivnguyen reviewed (approved):
- The "Sent" box works great. It's good that the completed requests disappear right after. 
- The hidden fields send in the correct request-id and tutor-id too (I wasn't able to test for the "Inbox" portion because they don't appear)

Some Bugs:
- The accepted requests of Inbox don't show up like the "Sent" box
- When clicking the profiles, the links don't lead to the tutor profiles anymore.
- It'll be good to make the "tutee/tutor" field be a hidden input, but @CalebNale can work with that for the next sprint.

Overall, there are no major bugs, so it could be merged for this sprint. Or, it could be saved for the next sprint. You can decide whether you would like to merge it

dtuckersuh committed:
Fix bug so accepted requests now show in Inbox and can now complete requests in Inbox

dtuckersuh committed:
Fix bug where links on homepage and requests page don't lead to correct profiles

dtuckersuh committed:
Fix status appearance on homepage

dtuckersuh committed:
Add 'Accepted' text to Completed requests

thivnguyen commented on tutor4points/core/templates/home.html:
```diff
@@ -44,14 +45,21 @@ <h2>Tutor Requests</h2>
-                                        data-toggle= "modal" data-target="#modal-decline-request">
-                                            Decline
-                                        </button>
+                                        {% if tutor_request.accepted %}
```
I don't see the "Complete" button appearing on the homepage for accepted requests on my end.

thivnguyen commented on tutor4points/core/templates/requests.html:
```diff
@@ -14,14 +14,26 @@
+            <input class="form-check-input" type="checkbox" value="" id="show-accepted" name="show-accepted"
+                onChange="this.form.submit()" {{accept_filter}}>
+            <label class="form-check-label" for="show-accepted">
+                Show Accepted
```
I think it'll be helpful to have 2 checkboxes as discussed in our last team meeting.  One for the "inbox" and one for the "sent".

thivnguyen reviewed (approved):
Everything looks good on the requests page! All the ids seem to be sent correctly, and forms are submitted. 

It'll be great if you could do 2 separate checkboxes as discussed in our last team meeting, one for "Inbox" and one for "Sent" so the user can filter separately.

The only bug I found was that the "complete" button does not appear for me on the homepage for any of the requests, so I was not able to test the functionality of that.

Overall, there is 1 minor bug that could be fixed quickly and one small change suggested. No major bugs so code could be merged.

cathyvu1 commented on tutor4points/core/views.py:
```diff
@@ -289,9 +290,52 @@ def requests(request, id):
+            userGivenTo.save()
+
+            currentRequest = TutorRequest.objects.get(pk=request.POST['request-id'])
+            currentRequest.completed = True
```
When a tutor marks a request as complete and submits a rating, the request will also disappear on the tutee's side before the tutee has submitted a rating (vice versa). I think we need to rethink how to check when a request is completed by both sides. One solution is to have a tutor_rating_completed and tutee_rating_completed for each request and then we can change completed to true when both tutor_rating_completed and tutee_rating_completed are true. Not sure if this is the best method though.

cathyvu1 commented on tutor4points/core/templates/requests.html:
```diff
@@ -14,14 +14,26 @@
+    <form method="post">
+    {% csrf_token %}
+        <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="" id="show-accepted" name="show-accepted"
```
Not a bug, but it would be nice to add styling to the checkbox so it matches the theme.

cathyvu1 reviewed (approved):
Overall functionality for accepted requests looks good.

CalebNale committed:
Removed rating from user profile

Removed all code related to the ratings old place on the user profile

CalebNale committed:
Changed form to include completed and paid

Additionally, automatically made checks for whether the user is a tutor or tutee.
Added one migration, adding a tutee_completed field to Request

CalebNale committed:
Solved /0 error, cleaned up code

Additionally: users must select paid and completed checkboxes, or else form does not go through
